### PR TITLE
Rename gtest to googletest

### PR DESCRIPTION
**Pull request by mboisson**
Renaming gtest to googletest everywhere

edit (boegel): this is done to avoid having easyconfigs for `googletest` under two different names...

### CODE REVIEW

**Comment by mboisson** [comment]
I don't think the failures are related to this PR. @boegel

File: easybuild/easyconfigs/g/googletest/googletest-1.10.0-GCCcore-9.3.0.eb [hunk 1]
```diff
@@ -16,11 +16,13 @@ builddependencies = [
...
 ]
-
+# build twice, once for static, once for shared libraries
+configopts = ['', ' -DBUILD_SHARED_LIBS=ON ']
 separate_build_dir = True
 
 sanity_check_paths = {
-    'files': ['lib/libgmock.a', 'lib/libgmock_main.a', 'lib/libgtest.a', 'lib/libgtest_main.a'],
+    'files': ['lib/libgmock.a', 'lib/libgmock_main.a', 'lib/libgtest.a', 'lib/libgtest_main.a',
+              'lib/libgmock.so', 'lib/libgmock_main.so', 'lib/libgtest.so', 'lib/libgtest_main.so'],
```
Comment (smoors) [inline easybuild/easyconfigs/g/googletest/googletest-1.10.0-GCCcore-9.3.0.eb:25]: @mboisson please change `.so` --> `SHLIB_EXT`?

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 11 out of 13 (8 easyconfigs in this PR)
node253.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz (broadwell), Python 2.7.5
See https://gist.github.com/79367807118d2fb3fdaea394a34c78f4 for a full test report.

**Comment by smoors** [comment]
@mboisson gsoap_2.8.48.zip seems not available on sourceforge anymore...

**Comment by mboisson** [comment]
@smoors hum ? There's no reference to `gsoap_2.8.48.zip` in this PR ?

**Comment by mboisson** [comment]
Oh... I see. It's a dependency.... but this is not part of this PR. Fixing the broken dependency is a separate issue, no ?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2429.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/79319d4e92dd33fdd267482b9210115b for a full test report.

**Comment by boegel** [comment]
Those test failures are not making any sense...

**Comment by boegel** [comment]
Ah, I see what the problem is... #11071 got merged, which lists a dep on `gtest`.

So this PR needs a sync with `develop` + fix in `Sumo` easyconfig.

**Comment by boegel** [comment]
@mboisson (or @bartoldeman) fixed with https://github.com/ComputeCanada/easybuild-easyconfigs/pull/12

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in this PR)
node2429.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/b5591276548a9574c6d1035993f69cf7 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @mboisson!

**Pull request merged**